src/ray/gcs/gcs_server/test/gcs_server_rpc_test.cc

```diff
@@ -127,6 +127,47 @@ class GcsServerTest : public ::testing::Test {
     AsyncCall(call_function, timeout_ms_);
   }
 
+  void TestRegisterNodeInfo(const rpc::RegisterNodeInfoRequest &request) {
+    auto call_function = [this, request](std::promise<bool> &promise) {
+      client_->RegisterNodeInfo(
+          request,
+          [&promise](const Status &status, const rpc::RegisterNodeInfoReply &reply) {
+            RAY_CHECK_OK(status);
+            promise.set_value(true);
+          });
+    };
+    AsyncCall(call_function, timeout_ms_);
+  }
+
+  void TestUnregisterNodeInfo(const rpc::UnregisterNodeInfoRequest &request) {
+    auto call_function = [this, request](std::promise<bool> &promise) {
+      client_->UnregisterNodeInfo(
+          request,
+          [&promise](const Status &status, const rpc::UnregisterNodeInfoReply &reply) {
+            RAY_CHECK_OK(status);
+            promise.set_value(true);
+          });
+    };
+    AsyncCall(call_function, timeout_ms_);
+  }
+
+  void GetAllNodesInfo(std::vector<rpc::GcsNodeInfo> &node_infos) {
```

Comment:
It is more better to define this method as `std::vector<rpc::GcsNodeInfo> GetAllNodesInfo()`.

Reply:
```
std::vector<rpc::GcsNodeInfo> node_infos;
auto call_function = [this, request, &node_infos]() {
  auto promise = std::promise<bool>();
  auto future = promise.get_future();
  client_->GetAllNodesInfo(
          request, [p = std::move(promise), &node_infos](const Status &status,
                                           const rpc::GetAllNodesInfoReply &reply) mutable  {
            RAY_CHECK_OK(status);
            for (int index = 0; index < reply.node_infos_size(); ++index) {
              node_infos.push_back(reply.node_infos(index));
            }
            p->set_value(true);
          });
  return future;
};
WaitReady(call_function, timeout_ms_);
return node_infos;
```

Reply:
fixed